tenebrae-press/cc-lcp-editor
# Implement strict typing via @tenebrae-press/lancer-types

Needs MAJOR testing. Touches nearly every editor/interface, should fix a lot of simple issues but may introduce others when loading data from things like No Room for a Wallflower LCPs.

Testing Process:

1. Open master (https://editor.tenebrae.press) and this branch
2. Load LCP containing content for testing
3. Compare content inside of editor, creating issues for unexpected different content with the prefix "Input Error:"
4. Export the LCP from both editors. Create issues from errors in this process with the prefix "Runtime Export Error:"
5. Unzip the contents into two separate folders. Compare each `.json` file. Create issues from unexpected differences/issues with the prefix "Data Export Error:"

**Check boxes below when testing is completed on a given vertical using the guide above.**
- [x] LCP Manifest 
- [x] Manufacturers
- [x] Frames
- [x] Weaponry
- [ ] Weapon Mods
- [x] Systems
- [ ] Backgrounds
- [ ] Environments
- [x] Pilot Gear
- [ ] Reserves
- [ ] Sitreps
- [ ] Statuses
- [x] Talents
- [ ] Tables
- [ ] NPC Classes
- [ ] NPC Templates
